packages/frontend/src/features/sqlRunner/components/visualizations/BarChartConfiguration.tsx

```diff
@@ -1,65 +1,414 @@
-import { Stack, Title } from '@mantine/core';
+import {
+    DEFAULT_AGGREGATION,
+    DimensionType,
+    type SqlTransformBarChartConfig,
+    type XLayoutOptions,
+    type YLayoutOptions,
+} from '@lightdash/common';
+import { Badge, Box, Button, Group, Select, Stack } from '@mantine/core';
 import debounce from 'lodash/debounce';
-import { type FC } from 'react';
+import { useState, type FC } from 'react';
+import { Config } from '../../../../components/VisualizationConfigs/common/Config';
 import { EditableText } from '../../../../components/VisualizationConfigs/common/EditableText';
 import {
     setSeriesLabel,
     setXAxisLabel,
+    setXAxisReference,
+    setYAxisAggregation,
     setYAxisLabel,
+    setYAxisReference,
 } from '../../store/barChartSlice';
 import { useAppDispatch, useAppSelector } from '../../store/hooks';
+import { TableFieldIcon } from '../TableFields';
+import { BarChartAggregationConfig } from './BarChartAggregationConfig';
 
 const DEBOUNCE_TIME = 500;
 
-const BarChartConfiguration: FC = ({}) => {
+type Axes = SqlTransformBarChartConfig;
+
+// TODO: add y field feature
+// const YFieldAxisConfigForm: FC<{
```

Comment:
Still need this commented section?

Reply:
Going to remove the code but keep the todo, sounds good?

Reply:
You can keep the code there too if it helps you. I was just checking